[macOS] Resolve all compute-kind-widget-generated WPT test failures when form control refresh enabled

lilyspiniolas opened the pull request:
#### fd4319719a1df49b96e93491ba4180684c7f23f2
<pre>
[macOS] Resolve all compute-kind-widget-generated WPT test failures when form control refresh enabled
<a href="https://bugs.webkit.org/show_bug.cgi?id=294368">https://bugs.webkit.org/show_bug.cgi?id=294368</a>
<a href="https://rdar.apple.com/153152167">rdar://153152167</a>

Reviewed by Aditya Keerthi and Abrar Rahman Protyasha.

Remove unnecessary border style adjustments when form control refresh
is enabled. All compute-kind-widget-generated WPT tests pass on macOS when
the setting is enabled as a result. Adjusted the padding of buttons to
compensate for the updated border values.

* Source/WebCore/rendering/RenderTheme.cpp:
(WebCore::RenderTheme::controlBorder const):
(WebCore::RenderTheme::adjustButtonOrCheckboxOrColorWellOrInnerSpinButtonOrRadioStyle const):
* Source/WebCore/rendering/RenderTheme.h:
* Source/WebCore/rendering/adwaita/RenderThemeAdwaita.cpp:
(WebCore::RenderThemeAdwaita::controlBorder const):
* Source/WebCore/rendering/adwaita/RenderThemeAdwaita.h:
* Source/WebCore/rendering/cocoa/RenderThemeCocoa.h:
* Source/WebCore/rendering/cocoa/RenderThemeCocoa.mm:
(WebCore::RenderThemeCocoa::controlBorder const):
(WebCore::RenderThemeCocoa::adjustColorWellStyleForVectorBasedControls const):
(WebCore::RenderThemeCocoa::adjustButtonStyleForVectorBasedControls const):
(WebCore::RenderThemeCocoa::adjustMenuListButtonStyleForVectorBasedControls const):
* Source/WebCore/rendering/mac/RenderThemeMac.h:
* Source/WebCore/rendering/mac/RenderThemeMac.mm:
(WebCore::RenderThemeMac::controlBorder const):

Canonical link: <a href="https://commits.webkit.org/296129@main">https://commits.webkit.org/296129@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/c83f59edf7bccac5e7c05e95d9cce66812ad8b4d

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/107480 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/27162 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/17570 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/112694 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/58016 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/109444 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/27847 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/35663 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/81586 "Passed tests") | [⏳ 🧪 win-tests](https://ews-build.webkit.org/#/builders/Win-Tests-EWS "Waiting to run tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/110409 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/22055 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/96866 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/61971 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/21492 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/15007 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/57460 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/91421 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/15038 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/115795 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/34547 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/25455 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/90624 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/34922 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/93122 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/90370 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/23038 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/35285 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/13059 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/30294 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/34468 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/40014 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/34214 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/37569 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/35875 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/e566fc6445cdb824d78f02ade68bdaef6cd1e58c. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/46643

aprotyas commented:
Do the WPTs require rebaselining?

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/3c9a96ee69807518f30c23cee087f2b2c0454e68. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/46643

lilyspiniolas commented:
> Do the WPTs require rebaselining?

They will eventually, but we only pass the remaining failures when the form control refresh setting is enabled.

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/b2d681b2f72ab686aef262588de91c4bc94dfb20. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/46643

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/18ccb5a8aaf990ebe5024d1de8c8b1ea394cca74. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/46643

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/c83f59edf7bccac5e7c05e95d9cce66812ad8b4d. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/46643

lilyspiniolas committed:
[macOS] Resolve all compute-kind-widget-generated WPT test failures when form control refresh enabled
https://bugs.webkit.org/show_bug.cgi?id=294368
rdar://153152167

Reviewed by Aditya Keerthi and Abrar Rahman Protyasha.

Remove unnecessary border style adjustments when form control refresh
is enabled. All compute-kind-widget-generated WPT tests pass on macOS when
the setting is enabled as a result. Adjusted the padding of buttons to
compensate for the updated border values.

* Source/WebCore/rendering/RenderTheme.cpp:
(WebCore::RenderTheme::controlBorder const):
(WebCore::RenderTheme::adjustButtonOrCheckboxOrColorWellOrInnerSpinButtonOrRadioStyle const):
* Source/WebCore/rendering/RenderTheme.h:
* Source/WebCore/rendering/adwaita/RenderThemeAdwaita.cpp:
(WebCore::RenderThemeAdwaita::controlBorder const):
* Source/WebCore/rendering/adwaita/RenderThemeAdwaita.h:
* Source/WebCore/rendering/cocoa/RenderThemeCocoa.h:
* Source/WebCore/rendering/cocoa/RenderThemeCocoa.mm:
(WebCore::RenderThemeCocoa::controlBorder const):
(WebCore::RenderThemeCocoa::adjustColorWellStyleForVectorBasedControls const):
(WebCore::RenderThemeCocoa::adjustButtonStyleForVectorBasedControls const):
(WebCore::RenderThemeCocoa::adjustMenuListButtonStyleForVectorBasedControls const):
* Source/WebCore/rendering/mac/RenderThemeMac.h:
* Source/WebCore/rendering/mac/RenderThemeMac.mm:
(WebCore::RenderThemeMac::controlBorder const):

Canonical link: https://commits.webkit.org/296129@main
